### PR TITLE
Microlearning

### PR DESCRIPTION
**Pull request by chrisdugne**
### Result and observation

<p align="center">
  
</p>

- [ ] **Breaking changes ?**  
       If checked, what have you broken ?

- [ ] **Extra dependency ?**
      If checked, Which extra dependency did you add ? (name, purpose, link ...).

###
 Testing Strategy

- [x] Already covered by tests
- [x] Manual testing
- [x] Unit testing

### CODE REVIEW

**Comment by chrisdugne** [comment]
[Mobile - ETQU je veux pouvoir faire un cours microlearning](https://trello.com/c/3hAJP6Ar/1183-mobile-etqu-je-veux-pouvoir-faire-un-cours-microlearning)

File: src/components/catalog-item.js [hunk 1]
```diff
@@ -112,7 +112,7 @@ const CatalogItem = ({
   section
 }: Props) => {
   const mode: CardDisplayMode = displayMode ? displayMode : CARD_DISPLAY_MODE.COVER;
-
+  const isCourse = type === 'learner';
```
Comment (VincentCATILLON) [inline src/components/catalog-item.js]: use constant

File: src/components/level-end.js [hunk 2]
```diff
@@ -177,10 +180,20 @@ class LevelEnd extends React.PureComponent<Props> {
       /{{levelName}}/g,
       levelUnlockedName
     );
+
+    const nextLabel = contentType === 'level' ? translations.nextLevel : translations.nextChapter;
```
Comment (VincentCATILLON) [inline src/components/level-end.js]: use constant

File: src/components/level-end.js [hunk 3]
```diff
@@ -177,10 +180,20 @@ class LevelEnd extends React.PureComponent<Props> {
...
+    const retryLabel =
+      contentType === 'level' ? translations.retryLevel : translations.retryChapter;
```
Comment (VincentCATILLON) [inline src/components/level-end.js]: use constant

File: src/components/text.js [hunk 4]
```diff
@@ -10,10 +10,12 @@ type Props = {|
...
-    {children}
-  </TextBase>
-);
+const Text = ({testID, children, style}: Props) => {
+  return (
+    <TextBase style={style} testID={testID}>
+      {children}
+    </TextBase>
+  );
+};
```
Comment (VincentCATILLON) [inline src/components/text.js]: 🤔 ?

File: src/layer/data/cards.js [hunk 5]
```diff
@@ -59,15 +74,20 @@ export const updateChapterCardAccordingToCompletion = (
   completion: Completion,
   chapterCard: ChapterCard
 ): ChapterCard => {
+  const config = getConfig({
+    ref: 'microlearning',
```
Comment (VincentCATILLON) [inline src/layer/data/cards.js]: use constant please

File: src/layer/data/cards.js [hunk 6]
```diff
@@ -78,18 +98,19 @@ const refreshDisciplineCard = async (disciplineCard: DisciplineCard): Promise<Di
...
-  return updateDisciplineCardDependingOnCompletion(levelCompletions, disciplineCard);
+  return updateDisciplineCardDependingOnCompletion(latestCompletions, disciplineCard);
 };
 
 const refreshChapterCard = async (chapterCard: ChapterCard): Promise<ChapterCard> => {
-  const completion: Completion = {stars: chapterCard.stars, current: chapterCard.completion};
-  const completionKey = buildCompletionKey('learner', chapterCard.moduleRef);
-  const storedCompletion = await AsyncStorage.getItem(completionKey);
+  const cardCompletion: Completion = {stars: chapterCard.stars, current: chapterCard.completion};
+  const completionKey = buildCompletionKey('microlearning', chapterCard.ref);
```
Comment (VincentCATILLON) [inline src/layer/data/cards.js]: use constant please

File: src/layer/data/cards.js [hunk 7]
```diff
@@ -78,18 +98,19 @@ const refreshDisciplineCard = async (disciplineCard: DisciplineCard): Promise<Di
...
+  const latestCompletion = await AsyncStorage.getItem(completionKey);
```
Comment (VincentCATILLON) [inline src/layer/data/cards.js]: this should be exposed by an abstract helper called `storage` but not the deal of this PR :)

File: src/layer/data/core.test.js [hunk 8]
```diff
@@ -240,79 +258,89 @@ describe('Data Layer Core', () => {
...
 
-    it('should try to fetch disciplineBundle', () => {
+    it('should fetch chapter bundle fixture', () => {
+      jest.mock('../../modules/environment', () => ({
+        __E2E__: true
+      }));
+      const {fetchBundle} = require('./core');
+      const keys = Object.keys(chaptersBundle.chapters);
+      const result = fetchBundle(CONTENT_TYPE.CHAPTER, keys[0], 'en', token, host);
+      // @todo should be mocked
```
Comment (VincentCATILLON) [inline src/layer/data/core.test.js]: please remove this

File: src/components/catalog-item.js [hunk 9]
```diff
@@ -112,7 +112,7 @@ const CatalogItem = ({
   section
 }: Props) => {
   const mode: CardDisplayMode = displayMode ? displayMode : CARD_DISPLAY_MODE.COVER;
-  const isCourse = type === 'learner';
+  const isCourse = type === CONTENT_TYPE.LEVEL;
```
Comment (VincentCATILLON) [inline src/components/catalog-item.js]: ??? should be ENGINE.LEARNER

**Comment by VincentCATILLON** [comment]
Should be tested functionally

**Comment by VincentCATILLON** [comment]
Functional ok :)

**Pull request merged**